lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -779,7 +824,8 @@ selectTxHistory wid order conditions = do
         ((TxMetaWalletId ==. wid) : conditions) sortOpt
     let txids = map txMetaTxId metas
     (ins, outs) <- selectTxs txids
-    pure $ txHistoryFromEntity @t metas ins outs
+    return $ prunePendingDuplicate $ txHistoryFromEntity @t metas ins outs
```

Comment:
It's a bit sad that our design forces us to traverse the tx history three times in order to remove duplicate pending transactions that happened to also be in ledger.
I'll try to work around that in a next PR...